### PR TITLE
Touch ups to `Array`

### PR DESCRIPTION
**Pull request by jakirkham**
* Add return type to `asarray` (makes easy to use with other typed values)
* Include `asarray` in `pxd` for faster access within other Cython modules

### CODE REVIEW

**Review by madsbk — APPROVED** [review]
Looks good, thanks @jakirkham

File: python/kvikio/kvikio/_lib/arr.pyi [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2020-2021, NVIDIA CORPORATION. All rights reserved.
+# Copyright (c) 2020-2024, NVIDIA CORPORATION. All rights reserved.
```
Comment (madsbk) [inline python/kvikio/kvikio/_lib/arr.pyi:1]:
```suggestion
# Copyright (c) 2020-2021, NVIDIA CORPORATION. All rights reserved.
```

File: python/kvikio/kvikio/_lib/arr.pyx [hunk 2]
```diff
@@ -289,9 +289,10 @@ cdef inline Py_ssize_t _nbytes(Py_ssize_t itemsize,
         nbytes *= shape_mv[i]
     return nbytes
 
-cpdef asarray(obj):
+
+cpdef Array asarray(obj):
```
Comment (madsbk) [inline python/kvikio/kvikio/_lib/arr.pyx:293]: A docstring would be good

File: python/kvikio/kvikio/_lib/arr.pyx [hunk 3]
```diff
@@ -289,9 +289,18 @@ cdef inline Py_ssize_t _nbytes(Py_ssize_t itemsize,
...
+    """Coerce other objects to ``Array``. No-op for existing ``Array``s.
+
+    Args:
+        obj: Object exposing the Python buffer protocol or ``__cuda_array_interface__``
+
+    Returns:
+        Array: An instance of the ``Array`` class
+    """
```
Comment (jakirkham) [inline python/kvikio/kvikio/_lib/arr.pyx:301]: How does this look?

Comment (madsbk) [reply python/kvikio/kvikio/_lib/arr.pyx:301]: Looks good, thanks @jakirkham !

**Pull request merged**